pkg/api/validate/dynamic/dynamic.go

```diff
@@ -389,6 +389,11 @@ func (dv *dynamic) ValidateSubnets(ctx context.Context, oc *api.OpenShiftCluster
 			}
 		}
 
+		if ss.SubnetPropertiesFormat == nil ||
+			ss.SubnetPropertiesFormat.ProvisioningState != mgmtnetwork.Succeeded {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, s.Path, "The provided subnet '%s' is not in a Succeeded state", s.ID)
```

Comment:
nit: for future reference (unless you want to make this small change) `%q` is the same as `'%s'`

Reply:
Ack!